nsd.c

```diff
@@ -114,12 +114,68 @@ version(void)
 	fprintf(stderr, "%s version %s\n", PACKAGE_NAME, PACKAGE_VERSION);
 	fprintf(stderr, "Written by NLnet Labs.\n\n");
 	fprintf(stderr,
-		"Copyright (C) 2001-2006 NLnet Labs.  This is free software.\n"
+		"Copyright (C) 2001-2020 NLnet Labs.  This is free software.\n"
 		"There is NO warranty; not even for MERCHANTABILITY or FITNESS\n"
 		"FOR A PARTICULAR PURPOSE.\n");
 	exit(0);
 }
 
+static void
+setup_verifier_environment(void)
+{
+	size_t i;
+	int ret, ip4, ip6;
+	char *buf, host[NI_MAXHOST], serv[NI_MAXSERV];
+	size_t size, cnt = 0;
+
+	/* allocate large enough buffer to hold a list of all ip addresses.
+	   ((" " + INET6_ADDRSTRLEN + "@" + "65535") * n) + "\0" */
+	size = ((INET6_ADDRSTRLEN + 1 + 5 + 1) * nsd.verify_ifs) + 1;
+	buf = xalloc(size);
+
+	ip4 = ip6 = 0;
+	for(i = 0; i < nsd.verify_ifs; i++) {
+		ret = getnameinfo(
+			(struct sockaddr *)&nsd.verify_udp[i].addr.ai_addr,
+			nsd.verify_udp[i].addr.ai_addrlen,
+			host, sizeof(host), serv, sizeof(serv),
+			NI_NUMERICHOST | NI_NUMERICSERV);
+		if(ret != 0) {
+			log_msg(LOG_ERR, "error in getnameinfo: %s",
+				gai_strerror(ret));
+			continue;
+		}
+		buf[cnt++] = ' ';
+		cnt += strlcpy(&buf[cnt], host, size - cnt);
+		assert(cnt < size);
+		buf[cnt++] = '@';
+		cnt += strlcpy(&buf[cnt], serv, size - cnt);
+		assert(cnt < size);
+#ifdef INET6
+		if (nsd.verify_udp[i].addr.ai_family == AF_INET6 && !ip6) {
+			setenv("VERIFY_IPV6_ADDRESS", host, 1);
+			setenv("VERIFY_IPV6_PORT", serv, 1);
+			setenv("VERIFY_IP_ADDRESS", host, 1);
+			setenv("VERIFY_IP_PORT", serv, 1);
```

Comment:
In doc/README the environment variable is `VERIFY_PORT` instead of `VERIFY_IP_PORT`.  `VERIFY_PORT` is also used in `tpkg/credns-setup` and in `tpkg/credns.tdir` (although that also sets the variable in the main environment, which is probably the reason you didn't notice it).
I have no strong opinion whether this should be `VERIFY_PORT` or whether the other environment variables should be `VERIFY_IP_PORT`, but the number of occurrences of `VERIFY_IP_PORT` is much lower, so probably easier the change.

Reply:
Good catch! Changed it back to `VERIFY_PORT`. Probably friendlier to users who want to migrate from CreDNS to NSD4 too :+1: